nucypher/crypto/powers.py

```diff
@@ -205,20 +206,11 @@ class KeyPairBasedPower(CryptoPowerUp):
 
     def __init__(self, public_key: PublicKey = None, keypair: keypairs.Keypair = None):
         if keypair and public_key:
-            raise ValueError("Pass keypair or pubkey_bytes (or neither), but not both.")
+            raise ValueError("Pass keypair or public key (or neither), but not both.")
         elif keypair:
             self.keypair = keypair
         else:
-            # They didn't pass a keypair; we'll make one with the bytes or
-            # Umbral PublicKey if they provided such a thing.
             if public_key:
-                try:
-                    public_key = public_key.as_umbral_pubkey()
-                except AttributeError:
-                    try:
-                        public_key = PublicKey.from_compressed_bytes(public_key)
-                    except TypeError:
-                        public_key = public_key
```

Comment:
Can you help me understand the rationale for this change?

Reply:
This intermediate base class is now meant to support two different cryptosystems (umbral and ferveo)  we cannot assume or infer the type.  Might need a little refactoring here.... perhaps the replacing of default keypair classes with `NotImplmplemented`.

Reply:
It seems to me that changes to `KeyPairBasedPower` are outside of the scope of this PR. I don't think I have the depth to address them now. I could just revert these changes and document this change in an issue.

Reply:
💯